src/projects/pull-request/pull-request-space.html

```diff
@@ -0,0 +1,228 @@
+          // default is top
+          points.leftX = rectangleLeft.left - containerRect.left;
+          points.leftY = rectangleLeft.top - containerRect.top;
+
+          if (dy < dx) {
+            // From right to left
+            points.leftX += rectangleLeft.width;
+            points.leftY += (rectangleLeft.height / 2);
+          } else {
+            points.leftX += (rectangleLeft.width / 2);
+
+            if (rectangleLeft.top < rectangleRight.top) {
+              // From bottom to left
+              points.leftY += rectangleLeft.height;
+            }
+          }
+
+          return points;
+        }
+
+        /**
+         * Determine which rectangle is left from the other and based on that
+         * calculate the anchorpoints for the arrow
+         * @param  {BoundingClientRect} containerRect The rectangle of the surrounding containerRect
+         * @param  {Object} fromRect      The rectangle of the start point of the arrow
+         * @param  {Object} toRect        The rectangle of the end point of the arrow
+         * @return {Object}               From and to anchor points for the corresponding rectangles
+         */
+        function calculateCoordinates(containerRect, fromRect, toRect) {
+          var dy = Math.abs(toRect.top - fromRect.top);
+          var dx = Math.abs(toRect.left - fromRect.left);
+          var anchorFrom, anchorTo;
+
+          var fromX, fromY, toX, toY;
+
+          // Arrow points to the right
+          if (fromRect.left < toRect.left) {
+            var points = calculateAnchorPoints(fromRect, toRect, dx, dy, containerRect);
+
+            return {
+              fromX: points.leftX,
+              fromY: points.leftY,
+              toX: points.rightX,
+              toY: points.rightY
+            };
+            // Arrow points to the left, so swap the values to make sure the Arrow
+            // points in the correct direction
+          } else {
+            var points = calculateAnchorPoints(toRect, fromRect, dx, dy, containerRect);
+
+            return {
+              fromX: points.rightX,
+              fromY: points.rightY,
+              toX: points.leftX,
+              toY: points.leftY
+            };
+          }
+        }
+
+        function drawArrow(ctx, fromX, fromY, toX, toY) {
+          ctx.beginPath();
+          ctx.moveTo(fromX, fromY);
+          ctx.lineTo(toX, toY);
+
+          // Draw the arrow point
+          // http://stackoverflow.com/a/6333775/2761676
+          var angle = Math.atan2(toY - fromY, toX - fromX);
+          ctx.moveTo(toX, toY);
+          ctx.lineTo(toX - 10 * Math.cos(angle - Math.PI / 6), toY - 10 * Math.sin(angle - Math.PI / 6));
+          ctx.moveTo(toX, toY);
+          ctx.lineTo(toX - 10 * Math.cos(angle + Math.PI / 6), toY - 10 * Math.sin(angle + Math.PI / 6));
+
+          ctx.closePath();
+          ctx.stroke();
+        }
+
+        // Recalc the canvas if we are currently dragging
+        return function() {
```

Comment:
I'd like to suggest not to use higher order functions with named inner functions for sake of readability. I see `_recalcCanvas` called from line 64, then scroll to the start of its definition a couple of lines below that, but the actual function is not created until this line (152). So I have to scroll all the way down, keep track of indentation and ignore functions inbetween that attract my attention in order to get to the core of this function. It would be much nicer if functions can be read somewhat in order from top to bottom.